app/models/prison.rb

```diff
@@ -35,6 +35,10 @@ def last_bookable_date(today = Time.zone.today)
     today + booking_window
   end
 
+  def bookable_date?(requested_date = Time.zone.today)
+    available_slots.any? { |slot| slot.on? requested_date }
```

Comment:
As a matter of style – and because of the way precedence works in Ruby – I think it's generally best to use parentheses anywhere that you're using the return value. It also provides an easy way to differentiate functional from imperative code.